content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,312 @@
+title: "Import and Run a python environment on Google cloud with Docker" 
+description: "Learn how to import and run a python environment on google cloud with docker"
+keywords: "docker, images, container, environments, virtual, Python, Jupyter, Jupyter notebook, environment, google cloud, cloud"
+weight: 2
+author: Diego Sanchez Perez
+draft: false
+aliases:
+
+
+---
+
+# Import and Run a python environment on Google cloud with Docker
+
+Take advantage of the versatility of containerized apps on Docker and the power of Google cloud to easily reproduce and collaborate on projects! 
+
+{{% summary %}}
+
+In this building block you will learn step by step how to replicate a Jupyter notebook Python environment from an existing project using Docker on an existing Google cloud instance. 
+
+{{% /summary %}}
+
+{{% warning %}}
+
+In order to be able to follow this building block you will need that the project you want to import already provides a both dockerfile and a docker-compose.yml file, with the adequate information to run the latter.
+
+{{% /warning %}}
+
+### Step 1: Install and Set up docker in a Google cloud instance
+
+Unless you have specified it otherwise, at the time of creating your instance Google cloud will use an image of the Debian Linux distribution as base OS for the instance. Accordingly you will have to follow the instructions' on how to install the docker engine for this specific Linux distribution. If you want to check a more detailed step-by-step version of these you can visit [the offical Docker documentation](https://docs.docker.com/engine/install/debian/), where you will also find instructions for installing docker on other Linux distributions and operative systems through the command line. If on the contrary you prefer to cut it straight to the chase, you can simply execute one by one the code lines below on your instance's command line to have Docker installed on your instance.
+
+{{% codeblock %}}
+```bash
+# Build a Docker image
+
+#1
+$ sudo apt-get remove docker docker-engine docker.io containerd runc
+
+#2
+$ sudo apt-get update
+
+#3
+$ sudo apt-get install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+
+#4
+$ sudo mkdir -p /etc/apt/keyrings
+
+#5
+$ curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+
+#6
+$ echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+#7
+$ sudo apt-get update
+
+#8
+$ sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+
+#9
+$ sudo docker run hello-world
+
+```
+{{% /codeblock %}}
+
+The last line of the code above (`$ sudo docker run hello-world`) is meant to act as a check to see if the installation was completed successfully. The command line output after running it will tell you if this was the case.
+

```

Comment:
We need to mention that on some Linux distributions you need to add your username to the docker group to be able to run docker commands without sudo. https://docs.docker.com/engine/install/linux-postinstall/